src/main/java/io/strimzi/kafka/metrics/KafkaMetricsCollector.java

```diff
@@ -125,6 +119,16 @@ static MetricFamilySamples convert(String name, String help, KafkaMetric metric,
                             throw new IllegalStateException("Unexpected duplicate key " + v1);
                         },
                         LinkedHashMap::new));
+
+        if (!(value instanceof Number)) {
+            // If the value is non-numeric, add it as a label and set the metric value to 1.0
+            LOG.info("*******Converting non-numeric metric {} with value {} to label", name, value);
+            sanitizedLabels.put("value", value.toString());
```

Comment:
can value ever be null?   To be defensive use String.valueOf(value) as that will never NPE on you.

Reply:
I'm not sure this is worthy of a log at info level.

Reply:
If the metric is measurable, the value is a primitive (double), so it can't be null, otherwise it can be an arbitrary type. While I don't expect null values, in this case it's easy to be safe so I agree with @k-wall, let's be defensive.

Reply:
Changed to String.valueOf(value)
LOG.info removed
`value` replaced with `name`, outputting the metric name for the label

Reply:
We need to use the name of the metric attribute. Here `name` is the full metric name.
For example, if the metric is `kafka_server_app_info_commit_id`, the label should just be `commit_id` and not the full name. The attribute name can be retrieved from the MetricName object.

Reply:
This is now refactored. Label has replaced full name and label